source/common/logger/ur_sinks.h

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: MIT
+
+#include <ostream>
+
+namespace sink
+{
+
+class Sink
+{
+public:
+    Sink() {
+        this->fd = stdout;
+    }
+
+    Sink(std::ostream &os)
+    {
+        if (os.rdbuf() == std::cerr.rdbuf()) {
+            this->fd = stderr;
+        }
+    }
+
+    ~Sink() = default;
+
+    template<typename... Args>
+    void log(const char *level, const char *format, Args&&... args) {
+        const int buf_size = 8192;
+        char buffer[buf_size];
+        int buf_size_left = buf_size;
+        unsigned position = 0;
+
+        appendToBuffer(position, buffer, buf_size_left, "[%s]:", level);
```

Comment:
This could become for example
```suggestion
        concatToBuffer(position, buffer, buf_size_left, '[', level, "]:");
```